From fc7923210c99f0e455a8377dfee515859d044a9d Mon Sep 17 00:00:00 2001
From: Muna Sinada <quic_msinada@quicinc.com>
Date: Wed, 6 Mar 2024 12:17:25 -0800
Subject: [PATCH] hostapd: DVLAN interface cleanup

During a network down operation, DVLAN (AP_VLAN) interface is not removed. In
case of DVLAN interface type, there is no need of remove each link
during network down operation.

Remove DVLAN type interface completely during a network down operation.

Signed-off-by: Muna Sinada <quic_msinada@quicinc.com>
Signed-off-by: Sidhanta Sahu <quic_sidhanta@quicinc.com>
---
 src/drivers/driver_nl80211.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

--- a/src/ap/ap_drv_ops.c
+++ b/src/ap/ap_drv_ops.c
@@ -629,7 +629,7 @@ int hostapd_if_remove(struct hostapd_dat
 		return -1;
 
 #ifdef CONFIG_IEEE80211BE
-	if (hapd->conf->mld_ap)
+	if (hapd->conf->mld_ap && type == WPA_IF_AP_BSS)
 		return hostapd_if_link_remove(hapd, type, ifname, hapd->mld_link_id);
 #endif /* CONFIG_IEEE80211BE */
 
